src/hotspot/share/opto/divnode.cpp

```diff
@@ -1517,16 +1517,16 @@ const Type* UModLNode::Value(PhaseGVN* phase) const {
 }
 
 Node* ModFNode::Ideal(PhaseGVN* phase, bool can_reshape) {
+  Node* super = CallLeafPureNode::Ideal(phase, can_reshape);
+  if (super != nullptr) {
+    return super;
+  }
+
   if (!can_reshape) {
     return nullptr;
   }
-  PhaseIterGVN* igvn = phase->is_IterGVN();
 
-  bool result_is_unused = proj_out_or_null(TypeFunc::Parms) == nullptr;
-  bool not_dead = proj_out_or_null(TypeFunc::Control) != nullptr;
-  if (result_is_unused && not_dead) {
-    return replace_with_con(igvn, TypeF::make(0.));
```

Comment:
Can we replace all the other usages of `ModFloatingNode::replace_with_con` by `TupleNode` and get rid of that method?

Reply:
Yes, that sounds good.